### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by yolile**
To address https://github.com/open-contracting-extensions/ocds_medicine_extension/pull/4#issuecomment-884440987 and update the JSON examples with the correct codes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,10 +79,10 @@ Clorhidrato de Bupivacaina Hiperbarica Inyectable     | clorhidrato de bupivacai
     "items": [
       {
         "id": "1",
-        "dosageForm": "injection",
-        "administrationRoute": "transdermal",
+        "dosageForm": "SOL",
+        "administrationRoute": "TRNSDERM",
         "container": {
-          "name": "blister",
+          "name": "blstrpk",
```
Comment (jpmckinney) [inline README.md]: Isn't the immediate container "ampollas" ('amp')? Like https://www.google.com/search?q=bupivacaine+hydrochloride+dextrose+injection+solution&tbm=isch

File: README.md [hunk 2]
```diff
@@ -79,10 +79,10 @@ Clorhidrato de Bupivacaina Hiperbarica Inyectable     | clorhidrato de bupivacai
     "items": [
       {
         "id": "1",
-        "dosageForm": "injection",
-        "administrationRoute": "transdermal",
+        "dosageForm": "SOL",
+        "administrationRoute": "TRNSDERM",
```
Comment (jpmckinney) [inline README.md]: An injection is never transdermal (like a nicotine patch). I think it's ISINJ (Injection, intraspinal) https://dailymed.nlm.nih.gov/dailymed/fda/fdaDrugXsl.cfm?setid=d2335583-1349-44ad-aac4-e9392533160c&type=display

File: README.md [hunk 3]
```diff
@@ -36,8 +36,8 @@ The strength is expressed as "100 MG/ML". The UN/CEFACT [Recommendation 20 – C
           "scheme": "UNSPSC",
           "uri": "https://apis.mercadopublico.cl/OCDS/data/productos/categoria/51161701"
         },
-        "dosageForm": "solution",
-        "administrationRoute": "nasal",
+        "dosageForm": "SOL",
+        "administrationRoute": "NASALINSTIL",
         "container": {
           "name": "jar",
```
Comment (jpmckinney) [inline README.md]: I think it's a vial, not a jar? As seen on https://www.google.com/search?q=%22ACETYLCYSTEINE%22+nebulizer+solution&tbm=isch

File: README.md [hunk 4]
```diff
@@ -36,8 +36,8 @@ The strength is expressed as "100 MG/ML". The UN/CEFACT [Recommendation 20 – C
           "scheme": "UNSPSC",
           "uri": "https://apis.mercadopublico.cl/OCDS/data/productos/categoria/51161701"
         },
-        "dosageForm": "solution",
-        "administrationRoute": "nasal",
+        "dosageForm": "SOL",
+        "administrationRoute": "NASALINSTIL",
```
Comment (jpmckinney) [inline README.md]: Since it's for a "NEBULIZAR" I think this is NASINHL (Inhalation, nasal).

File: README.md [hunk 5]
```diff
@@ -133,6 +133,8 @@ The fields, definitions and codelists used in this extension are based on the fo
...
 - The `container` codelist is a copy of the codes and titles from FHIR's [Medication Knowledge Package Type](https://terminology.hl7.org/CodeSystem/medicationknowledge-package-type/) codelist. Given that the terms are undefined in FHIR, the descriptions are copied from corresponding terms from the [EDQM Standard Terms database](https://standardterms.edqm.eu), reproduced with the permission of the European Directorate for the Quality of Medicines & HealthCare, Council of Europe (EDQM). The EDQM Standard Terms database is not a static list and content can change over time; the descriptions were retrieved on July 21, 2021.
 
+If you haven't adopted or can not map your `administrationRoute`, `dosageForm`, or `container` values to HL7, you can use your own codes. To make sure the data user understands these codes, you should name the code lists used in your publication policy document, and provide access to tables where the codes can be looked up.
+However, as the dosage forms and pack sizes tend to differ significantly across countries, it is strongly recommended to use standardized codes so the unit prices are easier to compare.
```
Comment (jpmckinney) [inline README.md]: Newline before following heading. Paragraph on one line.

Also, I think this paragraph belongs under Guidance, so I've edited it accordingly. We can put it after the second guidance paragraph.

```suggestion
This extension is designed to support [external reference pricing](https://en.wikipedia.org/wiki/External_reference_pricing). Dosage forms and container sizes differ significantly across countries, which makes comparison difficult. To ease comparison, the extension provides standardized codelists for the administration route, dosage form and immediate container, based on [Health Level Seven (HL7)](https://www.hl7.org), a set of international standards for health data. That said, if you haven't adopted and can't map your values to the HL7 codes, you may use your own codes. To allow a user to interpret your codes, you should describe the codelists, and how to find the definitions of codes, in your [publication policy](https://standard.open-contracting.org/latest/en/guidance/publish/#finalize-your-publication-policy).

```

**Review by jpmckinney — COMMENTED** [review]
For the Chile example, I'm not sure that the link matches the example. I don't see "NEBULIZAR" anywhere in the linked OCDS data.

The DNCP example has "VIAL", but if I look at LOTE8 at https://www.contrataciones.gov.py/licitaciones/convocatoria/391507-adquisicion-medicamentos-hospital-clinicas-1.html#itemSolicitados, the presentation is AMPOLLA.

We should fix VIAL, the link (#itemSolicitados not #pliego), and also add "(see lot 8)" or something, because it's otherwise very hard to find the corresponding text.

We can also link to the OCDS: https://contrataciones.gov.py/datos/api/v3/doc/ocds/record/ocds-03ad3f-391507

I can't tell where the "Technical specifications" and "Delivery presentation" are from.

**Comment by yolile** [comment]
> For the Chile example, I'm not sure that the link matches the example. I don't see "NEBULIZAR" anywhere in the linked OCDS data.

I think that the original, non-OCDS link for Chile was this one https://www.mercadopublico.cl/Procurement/Modules/RFB/DetailsAcquisition.aspx?qs=OE1kSVnLUBVxS5IkXPNLRQ== where you can indeed see "NEBULIZAR", maybe we should use this one instead? or both as Paraguay now?

> I can't tell where the "Technical specifications" and "Delivery presentation" are from.

I looked at this and found it in https://www.contrataciones.gov.py/licitaciones/convocatoria/391507-adquisicion-medicamentos-hospital-clinicas-1.html#pliego -> "Suministros requeridos - especificaciones técnicas" - LOTE N° 8 - ANESTESICOS LOCALES - 2, item 1

File: README.md (hunk 3 above)
Comment (yolile) [reply README.md]: Hmm the description says "frasco" which means a jar

Comment (jpmckinney) [reply README.md]: I have never seen this sold as a "jar". Individual 15-30 ml containers are more likely vials. It seems frasco is sometimes used to describe vials. Even when I search for a "bottle" of acetylcysteine, the picture is a vial.

**Review by jpmckinney — APPROVED** [review]
I made one commit to improve the flow of the example text, so that the OCDS links and navigation assistance come later.

**Comment by jpmckinney** [comment]
Actually, can we find the INN for the active ingredients? I'm looking for a source. Ideally we can add the URL to the readme / schema, if we find a good list.

**Comment by yolile** [comment]
this is the official and latest one https://www.who.int/publications/m/item/inn-rl-85 (as PDF here https://cdn.who.int/media/docs/default-source/international-nonproprietary-names-(inn)/rl85.pdf?sfvrsn=9a0014be_12&download=true)

**Comment by jpmckinney** [comment]
> this is the official and latest one https://www.who.int/publications/m/item/inn-rl-85 (as PDF here https://cdn.who.int/media/docs/default-source/international-nonproprietary-names-(inn)/rl85.pdf?sfvrsn=9a0014be_12&download=true)

~~I can't tell if each of those is complete or not.~~ Each of those is only a partial list/update. In the [Stem Book](https://www.who.int/teams/health-product-and-policy-standards/inn/stembook), I can find 'acetylcysteine' and 'bupivacaine'. ~~I think the standard works by describing how to compose larger compounds like bupivacaine hydrochloride (I haven't been able to confirm the latter yet).~~ https://github.com/open-contracting-extensions/ocds_medicine_extension/pull/6#issuecomment-885845929

**Comment by yolile** [comment]
I found this [document](https://cdn.who.int/media/docs/default-source/essential-medicines/2021-eml-expert-committee/applications-for-new-indications-for-existing-listed-medicines/i.13_n-acetylcysteine.pdf?sfvrsn=6945b088_4) about acetylcysteine, maybe it is too new?

**Comment by jpmckinney** [comment]
I wrote a command to download all the proposed/recommended lists, so that I can grep.

**Comment by jpmckinney** [comment]
I think the stem book is actually the best resource. I has stems (-caine, -steine) followed by INNs (bupivacaine, acetylcysteine). Annex 2 (General principles for guidance in devising international nonproprietary names for pharmaceutical substances) explains that acid salts are named like "{acid} {salt}" so "bupivacaine hydrochloride").

**Comment by jpmckinney** [comment]
Okay, I've updated to use and explain INNs. The examples now follow all our recommendations. Please review and merge.

**Comment by yolile** [comment]
> Okay, I've updated to use and explain INNs. The examples now follow all our recommendations. Please review and merge.

Are you sure that we should use the names in English? See, for example, page 20 of this [guidance](https://www.who.int/medicines/services/inn/FINAL_WHO_PHARM_S_NOM_1570_web.pdf?ua=1) or Annex 3 at this [document](https://www.who.int/medicines/services/inn/Radical_Book_2012.pdf) "Standardization of the Spanish version of INN "

**Comment by jpmckinney** [comment]
Hmm, I'm looking for this "cumulative list" that has official translations, but WHO really makes it a pain.

**Update:** These all seem restricted.

* https://mednet-communities.net/inn: "Access to MedNet is restricted and managed by the leader of each scientific community"
* https://extranet.who.int/tools/inn_global_data_hub/: "a formal mail is needed" according to https://extranet.who.int/tools/inn_global_data_hub/INN_Hub_User_API_V20.pdf
* https://extranet.who.int/tools/inn_online_application/: I think this is only to request an INN anyway, not to lookup an INN.

**Comment by jpmckinney** [comment]
acetylcysteine is in Recommended List 05 from 1965, which is a scanned PDF, which explains why I couldn't search for it earlier. In 1965, they weren't providing additional languages (except Latin), so the Spanish must be somewhere else.

bupivacaine is from RL07.

**Comment by jpmckinney** [comment]
Also, where is the guidance linked from? The Radical Book is here (newer 2015 version): https://www.who.int/teams/health-product-and-policy-standards/inn/inn-schemas

**Update:** Okay, the guidance is https://www.who.int/publications/m/item/guidance-on-the-use-of-inns (newer version) (fixed link)

**Comment by yolile** [comment]
Do you mean here https://www.who.int/publications/i/item/who-emp-qsm-2015-1 ?

**Comment by jpmckinney** [comment]
That list is only for radicals and groups (e.g. it doesn't have the INNs we need, which were already on the recommended lists in the 1960s). I also fixed my link in my previous comment.

**Comment by jpmckinney** [comment]
Okay, well I can find the INN terms like:

* https://pubchem.ncbi.nlm.nih.gov/compound/acetylcysteine
* https://pubchem.ncbi.nlm.nih.gov/compound/bupivacaine
* https://pubchem.ncbi.nlm.nih.gov/compound/bupivacaine-hydrochloride

The Spanish INN is in the list of "Depositor-Supplied Synonyms" for the first two.

For the acid salt, I can find both "bupivacaina clorhidrato" and "clorhidrato de bupivacaina" online. I haven't looked for Spanish guidance on how to express Spanish salts (the English guidance is linked in the readme).

**Comment by jpmckinney** [comment]
Anyway, maybe we just recommend the Latin? Having INNs in English, French, Spanish, Arabic, Chinese and Russian doesn't sound like fun for users to reconcile.

**Comment by yolile** [comment]
> Anyway, maybe we just recommend the Latin? Having INNs in English, French, Spanish, Arabic, Chinese and Russian doesn't sound like fun for users to reconcile.

Having to translate them to English doesn't sound like fun for the publishers either :( So maybe yes, recommend Latin is the best

**Comment by jpmckinney** [comment]
@yolile Done now. Please review and merge if okay.

**Pull request merged**